src/components/Course/CourseCaution.vue

```diff
@@ -39,27 +45,45 @@
 import { PropType, defineComponent } from 'vue';
 import CourseBaseTooltip from '@/components/Course/CourseBaseTooltip.vue';
 import store from '@/store';
-import { isPlaceholderCourse } from '@/utilities';
+import { isCourseConflict } from '@/global-firestore-data';
+import { isPlaceholderCourse, isCourseTaken } from '@/utilities';
 
 type CourseCautions = {
   readonly noMatchedRequirement: boolean;
   readonly typicallyOfferedWarning: readonly string[] | undefined;
   readonly isCourseDuplicate: boolean;
   readonly isPlaceholderWrongSemester: boolean;
+  readonly hasConflictRequirement: boolean;
 };
 
 const getCourseCautions = (
-  course: FirestoreSemesterCourse | FirestoreSemesterPlaceholder,
+  course: FirestoreSemesterCourse | FirestoreSemesterPlaceholder | CourseTaken,
   semesterIndex: number
 ): CourseCautions => {
   const {
     safeRequirementFulfillmentGraph,
     derivedCoursesData: { duplicatedCourseCodeSet, courseToSemesterMap },
   } = store.state;
+
+  const uniqueID = isCourseTaken(course) ? course.uniqueId : course.uniqueID;
```

Comment:
This is pretty confusing, we should standardize it to `uniqueId` or something in a future PR (so all 3 kinds of "courses" have the same key for unique id)

Reply:
Agreed! It took me a long time to figure out why `course.uniqueId` was not working only to discover the capitalization difference. Made an item: https://www.notion.so/courseplan/82bf355c4b7b49f99691eb9f3afb203c?v=5da9d646f3ef424bbceb1ca0e8e00cf8&p=f0868e5ead134bdf8acdd11405d7eb3f